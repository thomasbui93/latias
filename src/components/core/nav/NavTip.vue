<template functional>
  <router-link :to="props.path">
    <div class="nav-tip">
      <div class="nav-tip__background">
        <img :src="props.backgroundImage"/>
      </div>
      <div class="nav-tip__text">
        <span>{{props.name}}</span>
      </div>
    </div>
  </router-link>
</template>

<style lang="scss" scoped>
  $navTipHeight: 100px!default;

  .router-link-exact-active > .nav-tip .nav-tip__text {
    background-color: rgba(0,0,0,0.7)
  }

  .nav-tip {
    display: inline-block;
    width: $navTipHeight;
    height: $navTipHeight;
    position: relative;
    border-radius: 30px;
    overflow: hidden;
    margin: 0 .5rem;

    & > div {
      width: 100%;
      height: 100%;
      left: 0;
      top: 0;
    }

    &__background {
      position: relative;
      border-radius: 10px;
    }

    &__text {
      text-align: center;
      position: absolute;
      background-color: rgba(0,0,0,0.3);
      color: white;
      font-weight: bold;
      display: flex;
      justify-content: center;
      align-items: center;
    }
  }
</style>
