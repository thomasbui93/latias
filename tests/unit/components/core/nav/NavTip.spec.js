import { shallowMount, createLocalVue } from '@vue/test-utils';
import NavTip from '@/components/core/nav/NavTip.vue';
import VueRouter from 'vue-router';

const localVue = createLocalVue();
localVue.use(VueRouter);

describe('NavTip component', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(NavTip, {
      stubs: ['router-link', 'router-view'],
      propsData: {
        path: '/',
        backgroundImage: '/sample.jpg',
        name: 'Latias',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
