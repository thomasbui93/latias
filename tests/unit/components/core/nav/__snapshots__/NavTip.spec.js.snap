// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavTip component matches snapshot 1`] = `
<router-link-stub path="/" backgroundimage="/sample.jpg" name="Latias">
  <div class="nav-tip">
    <div class="nav-tip__background"><img></div>
    <div class="nav-tip__text"><span></span></div>
  </div>
</router-link-stub>
`;
